### PR TITLE
Migrate SeekBarPreferenceCompat to kotlin

### PR DESCRIPTION
**Pull request by lukstbit**
## Purpose / Description

Migrates the SeekBarPreferenceCompat.java file to kotlin following the kotlin migration guide. This finishes the migration for the `com.ichi2.preferences` package.

## How Has This Been Tested?

Ran the usual tests.

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/preferences/SeekBarPreferenceCompat.kt [hunk 1]
```diff
@@ -24,267 +24,248 @@
...
-    @SuppressWarnings("deprecation") // 5019 - onSetInitialValue
-    @Override
-    protected void onSetInitialValue(boolean restore, Object defaultValue) {
-        super.onSetInitialValue(restore, defaultValue);
-        mValue = getPersistedInt(mDefault);
-        if (restore) {
-            mValue = shouldPersist() ? getPersistedInt(mDefault) : 0;
+    @Suppress("DEPRECATION")
+    @Deprecated("Deprecated in Java")
+    override fun onSetInitialValue(restore: Boolean, defaultValue: Any) {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/preferences/SeekBarPreferenceCompat.kt:87]: bug: this should be `Any?`

File: AnkiDroid/src/main/java/com/ichi2/preferences/SeekBarPreferenceCompat.kt [hunk 2]
```diff
@@ -24,267 +24,248 @@
...
+import androidx.annotation.StringRes
+import androidx.appcompat.app.AlertDialog
+import androidx.preference.DialogPreference
+import androidx.preference.PreferenceDialogFragmentCompat
+import com.ichi2.anki.AnkiDroidApp
+import com.ichi2.ui.FixedTextView
+import com.ichi2.utils.KotlinCleanup
+
+@KotlinCleanup("fix IDE lint issues")
+class SeekBarPreferenceCompat : DialogPreference {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/preferences/SeekBarPreferenceCompat.kt:47]: I believe we can move some of the `onCreateDialogView` variables to `lateinit`

**Review by david-allison — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/preferences/SeekBarPreferenceCompat.kt (hunk 1 above)
Comment (lukstbit) [reply AnkiDroid/src/main/java/com/ichi2/preferences/SeekBarPreferenceCompat.kt]: Fixed!

File: AnkiDroid/src/main/java/com/ichi2/preferences/SeekBarPreferenceCompat.kt (hunk 2 above)
Comment (lukstbit) [reply AnkiDroid/src/main/java/com/ichi2/preferences/SeekBarPreferenceCompat.kt:47]: @david-allison I've added a `@KotlinCleanup` to `mSuffix: String?` which could be made a `lateinit var`, the others are `Int` primitives and can't be made lateinit.

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/preferences/SeekBarPreferenceCompat.kt:47]: Sounds good to me, thanks!

**Review by david-allison — APPROVED** [review]

**Review by mikehardy — APPROVED** [review]
works for me - thanks!

**Pull request merged**